ext/vscode/src/commands/down.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { IActionContext } from "@microsoft/vscode-azext-utils";
+import { localize } from '../localize';
+import { quickPickWorkspaceFolder } from '../utils/quickPickWorkspaceFolder';
+import { getAzDevTerminalTitle, pickAzureYamlFile, selectApplicationTemplate, showReadmeFile } from './cmdUtil';
+import { createAzureDevCli } from '../utils/azureDevCli';
+import { executeAsTask } from '../utils/executeAsTask';
+import { TelemetryId } from '../telemetry/telemetryId';
+import { AzureDevCliApplication } from '../views/workspace/AzureDevCliApplication';
+import { isTreeViewModel, TreeViewModel } from '../utils/isTreeViewModel';
+
+export async function down(context: IActionContext, selectedItem?: vscode.Uri | TreeViewModel): Promise<void> {
+    const selectedFile = isTreeViewModel(selectedItem) ? selectedItem.unwrap<AzureDevCliApplication>().context.configurationFile : selectedItem;
+
+    let folder: vscode.WorkspaceFolder | undefined = (selectedFile ? vscode.workspace.getWorkspaceFolder(selectedFile) : undefined);
+    if (!folder) {
+        folder = await quickPickWorkspaceFolder(context, localize('azure-dev.commands.cli.init.needWorkspaceFolder', "To run '{0}' command you must first open a folder or workspace in VS Code", 'down'));
+    }
+
+    const azureCli = await createAzureDevCli(context);
+    let command = azureCli.commandBuilder
+        .withArg('down');
+    let workingDir = folder.uri;
+
+    const azureYamlFile = selectedFile ?? await pickAzureYamlFile(context);
+    if (azureYamlFile) {
+        // Workspace has already been initialized, no need to specify a template
+        workingDir = vscode.Uri.file(path.dirname(azureYamlFile.fsPath));
+    } else {
+        const templateUrl = await selectApplicationTemplate(context);
+        command = command.withNamedArg('-t', {value: templateUrl, quoting: vscode.ShellQuoting.Strong});
+    }
+
+    // TODO: Prompt user to confirm deletion (as well as for Key Vaults?).
```

Comment:
Alternatively, we could execute the command in terminal and have the CLI do the warning/prompting.

Reply:
That's what's currently happening. Personally I'd kinda prefer the prompts and such to happen within VSCode, and then use flags like `--force` and `--purge` to prevent prompts in the CLI. But I can see some users preferring each approach; I'm wondering if a setting is in order. In any case, I think it's probably best addressed as a followup PR.